src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -5205,6 +5205,7 @@ public override void PostVisit(TableNode node)
                     }
                     else if (isChildTypeAllowedInTable && childType.CoercesTo(exprType))
                     {
+                        exprType = DType.UnionWithCoercion(exprType, childType);
```

Comment:
This should be done based on a feature, so that we don't update Power Apps until it's ready to take this change.